packages/attribute-slicer-powerbi/src/AttributeSlicerVisual.ts

```diff
@@ -305,18 +305,24 @@ export default class AttributeSlicer extends VisualBase {
                     delete this.loadDeferred;
                 }
 
-                const columnName = ldget(dv, "categorical.categories[0].source.queryName");
+                const columnNames: Array<String> = [];
+                _.forOwn(ldget(dv, "categorical.categories"), (value, key: any) => {
+                    columnNames.push(value.source.queryName);
+                });
 
                 // Only clear selection IF
                 // We've already loaded a dataset, and the user has changed the dataset to something else
-                if (this.currentCategory && (!columnName || this.currentCategory !== columnName)) {
+                if (this.currentCategory && (!columnNames || !_.isEqual(this.currentCategory, columnNames)))  {
                     // This will really be undefined behaviour for pbi-stateful because this indicates the user changed datasets
                     log("Clearing Selection, Categories Changed");
-                    pbiState.selectedItems = [];
+                    if (!_.isEqual(pbiState.selectedItems, [])) {
+                        pbiState.selectedItems = [];
+                        this._onSelectionChangedDebounced([]);
```

Comment:
note that this line is new.  Without it i noticed that the selection token was removed by the actual selection was not (other visuals were still filtered, and if the category was changed back to the original the selection token would return as the category was still selected).